README.md

```diff
@@ -395,6 +395,20 @@ The second is the interface's supported property indices/names. By default, the
 
 If the getter function always returns a constant value for unsupported properties, webidl2js also offers a non-standard extended attribute `[WebIDL2JSValueAsUnsupported]` (documented below) that would simply call the getter function to check if a property index/name is supported, so that `supportsPropertyIndex`/`supportsPropertyName` would not need to be implemented separately. However, when using the extended attribute, be very sure that the value specified in the attribute is returned *if and only if* the property is unsupported.
 
+### Iterables
+
+For synchronous value iterable declarations, there is no need to add implementation-class code: they will be automatically generated based on the indexed property getter and `length` property.
+
+For synchronous pair iterable declarations, the implementation class needs to implement the `[Symbol.iterator]()` property, returning an iterable of `[key, value]` pairs. These can be impls; the generated code will convert them into wrappers as necessary.
+
+### Async iterables
+
+[Asynchronous iterable declarations](https://heycam.github.io/webidl/#idl-async-iterable) require the implementation class to implement the following symbol-named methods, corresponding to algorithms from the Web IDL specification. The `utils` object below refers to the default export from the generated utilities file `utils.js`.
+
+- `utils.asyncIteratorNext`: corresponds to the [get the next iteration result](https://heycam.github.io/webidl/#dfn-get-the-next-iteration-result) algorithm, and receives a single argument containing an instance of the generated async iterator. For pair asynchronous iterables, the return value must be a `[key, value]` pair array, or `undefined` to signal the end of the iteration. For value asynchronous iterables, the return value must be the value, or `undefined` to signal the end of the iteration.
```

Comment:
Hmm, this seems kind of bad. Streams can contain undefined values. Probably a spec bug?

Reply:
Web IDL doesn't really have a native “undefined” type though, so when the spec says "It must return a `Promise` that either rejects, resolves with undefined to signal the end of the iteration" in the IDL section (rather than the ES bindings section) I guess it's a bit unclear what exactly this "undefined" refers to. The conversion algorithms for [`any`](https://heycam.github.io/webidl/#es-any) says "an `object` reference to a special object that represents the ECMAScript **undefined** value".

Reply:
I'll push a new commit that follows https://github.com/heycam/webidl/pull/885